src/d3a/models/state.py

```diff
@@ -43,131 +46,249 @@
 # - If a device has no state, maybe it doesn't need its own appliance class either
 
 
-class PVState:
+class StateInterface(metaclass=ABCMeta):
+    """Interface containing methods that need to be defined by each State class."""
+
+    @abstractmethod
+    def get_state(self) -> Dict:
+        """Return the current state of the device."""
+        return {}
+
+    @abstractmethod
+    def restore_state(self, state_dict: Dict):
+        pass
+
+    @abstractmethod
+    def delete_past_state_values(self, time_slot):
+        pass
+
+    def __str__(self):
+        return self.__class__.__name__
+
+
+class ConsumptionState(StateInterface):
+    """State for devices that can consume energy."""
+
     def __init__(self):
+        super().__init__()
+        # Energy that the load wants to consume (given by the profile or live energy requirements)
+        self._desired_energy_Wh = {}
+        # Energy that the load needs to consume. It's reduced when new energy is bought
+        self._energy_requirement_Wh = {}
+        self._total_energy_demanded_Wh = 0
+
+    def get_state(self) -> Dict:
+        """Return the current state of the device. Extends super implementation."""
+        state = super().get_state()
+        consumption_state = {
+            "desired_energy_Wh": convert_pendulum_to_str_in_dict(self._desired_energy_Wh),
+            "total_energy_demanded_Wh": self._total_energy_demanded_Wh}
+        # The inherited state should not have keys with the same name (to avoid overwriting them)
+        conflicting_keys = state.keys() & consumption_state.keys()
+        assert not conflicting_keys, f"Conflicting state values found for {conflicting_keys}."
```

Comment:
We should not rely on assertions in our production code, a custom exception would work instead

Reply:
This is not supposed to be a check for production: it should break the system during the development stages.

Reply:
To reiterate what @fievelk  says, I am a big fan of assertions in production code, especially when debugging. When the code stumbles upon a condition that will definitely crash the system, it is much more easily debugged and reproduced to check why the assertion fails than trying to find the root cause of the issue through logs. 
For instance, this is a perfect example of a programming error (the parent classes should not have the same state keys as the child classes). It is much easier even for the developer to get notified immediately that the implementation is incorrect during development, rather than let it slide and have runtime errors (or, for this case, errors in the values reported in the UI). 
To the exceptions / assertions discussion, an assert is a type of an exception (AssertionError) meant exactly for this purpose. It just has a convenient syntax.